scripts/utils/dtdoctor_analyzer.py

```diff
@@ -0,0 +1,192 @@
+(e.g ./build/zephyr/edt.pickle) and a symbol that appeared in the build error
+message (e.g. __device_dts_ord_123).
+
+Example usage:
+
+./scripts/utils/dtdoctor_analyzer.py \
+    --edt-pickle ./build/zephyr/edt.pickle \
+    --symbol __device_dts_ord_123
+
+"""
+
+import argparse
+import os
+import pickle
+import re
+import sys
+
+sys.path.insert(0, os.path.join(os.path.dirname(__file__), "..", "dts", "python-devicetree", "src"))
+
+from tabulate import tabulate
+
+
+def parse_args():
+    parser = argparse.ArgumentParser(allow_abbrev=False)
+    parser.add_argument("--edt-pickle", required=True)
+    parser.add_argument("--symbol", required=True)
+    return parser.parse_args()
+
+
+def load_edt(path):
+    with open(path, "rb") as f:
+        return pickle.load(f)
+
+
+def setup_kconfig():
+    scripts_dir = os.path.abspath(os.path.join(os.path.dirname(__file__), ".."))
+    sys.path.extend([scripts_dir, os.path.join(scripts_dir, "kconfig")])
+    import kconfiglib
+
+    kconf = kconfiglib.Kconfig(os.path.join(os.environ.get("ZEPHYR_BASE"), "Kconfig"), warn=False)
+    return kconf, kconfiglib
+
+
+def format_node(node):
+    if node.labels:
+        return f"{node.labels[0]}: {node.path}"
+    return node.path
+
+
+def find_kconfig_deps(kconf, kconfiglib, dt_has_symbol):
+    """
+    Find all Kconfig symbols that depend on the provided DT_HAS symbol.
+    """
+    prefix = os.environ.get("CONFIG_", "CONFIG_")
+    target = f"{prefix}{dt_has_symbol}"
+    deps = set()
+
+    def collect_syms(expr):
+        # Recursively collect all symbol names in the expression tree except the target
+        for item in kconfiglib.expr_items(expr):
+            if not isinstance(item, kconfiglib.Symbol):
+                continue
+            sym_name = f"{prefix}{item.name}"
+            if sym_name != target:
+                deps.add(sym_name)
+
+    for sym in getattr(kconf, "unique_defined_syms", []):
+        for node in sym.nodes:
+            # Check dependencies
+            if node.dep is None:
+                continue
+            dep_str = kconfiglib.expr_str(
+                node.dep,
+                lambda sc: f"{prefix}{sc.name}" if hasattr(sc, 'name') and sc.name else str(sc),
+            )
+            if target in dep_str:
+                collect_syms(node.dep)
+
+            # Check selects/implies
+            for attr in ["orig_selects", "orig_implies"]:
+                for value, _ in getattr(node, attr, []) or []:
+                    value_str = kconfiglib.expr_str(value, str)
+                    if target in value_str:
+                        collect_syms(value)
+
+    return deps
+
+
+def handle_enabled_node(node):
+    """
+    Handle diagnosis for an enabled DT node (linker error, one or more Kconfigs might be gating
+    the device driver).
+    """
+    lines = []
+    lines.append(
+        f"'{format_node(node)}' is enabled but no driver appears to be available for it.\n"
+    )
```

Comment:
Will it be faster (One `append` call is reduced)?
```suggestion
    lines = [f"'{format_node(node)}' is enabled but no driver appears to be available for it.\n"]
```